yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,162 @@
+        m = re.match('(?P<vc>[^,]+),(?P<ac>[^,]+)', codec_str)
+        if m:
+            vcodec = m.group("vc")
+            acodec = m.group("ac")
+
+        return {
+            'url': info.get('url'),
+            'fps': int_or_none(info.get('frameRate')),
+            'width': int_or_none(info.get('width')),
+            'height': int_or_none(info.get('height')),
+            'vcodec': vcodec,
+            'acodec': acodec,
+            'tbr': float_or_none(info.get('avgBitrate'))
+        }
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
+
+        webpage = self._download_webpage(url, video_id)
+        json_all = self._search_json(r'window.videoInfo\s*=\s*', webpage, 'videoInfo', video_id)
+
+        video_info = json_all['currentVideoInfo']
+        playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+        video_inner_id = traverse_obj(json_all, ('currentVideoInfo', 'id'))
+
+        ext = determine_ext(video_info.get('fileName'), 'mp4')
+
+        formats = []
+        for video in traverse_obj(playjson, ('adaptationSet', 0, 'representation')):
+            fmt = AcFunVideoIE.parse_format(video)
+            formats.append({**fmt, 'ext': ext})
+
+        self._sort_formats(formats)
+
+        video_list = json_all['videoList']
+        p_idx, video_info = [(idx + 1, v) for (idx, v) in enumerate(video_list)
+                             if v['id'] == video_inner_id
+                             ][0]
+
+        title = json_all['title']
+        if len(video_list) > 1:
+            title = f"{title} P{p_idx:02d} {video_info['title']}"
+
+        return {
+            'id': video_id,
+            'title': title,
+            'duration': float_or_none(video_info.get('durationMillis'), 1000),
+            'timestamp': int_or_none(video_info.get('uploadTime'), 1000),
+            'formats': formats,
+            'http_headers': {
+                'Referer': url,
+            },
+        }
+
+
+class AcFunBangumiIE(InfoExtractor):
+    IE_NAME = 'AcFunBangumiIE'
+    _VALID_URL = r'''(?x)
+                    https?://(?:www\.acfun\.cn/bangumi/)
+                        (?P<id>aa[_\d]+)
+                        (?:\?ac=(?P<ac_idx>\d+))?
+                    '''
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/bangumi/aa6002917',
+        'info_dict': {
+            'id': 'aa6002917',
+            'title': '租借女友 第1话 租借女友',
+            'duration': 1467,
+            'timestamp': 1594432800,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/bangumi/aa6002917',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.acfun.cn/bangumi/aa6002917_36188_1745457?ac=2',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
```

Comment:
same as https://github.com/yt-dlp/yt-dlp/pull/4228/files#r914718187. You can also group `ac_idx` from here, regex will return `None` if it's not found

Reply:
ok